### DISCUSSION TITLE
Newbie: how do I do a simple byte serialization of a string schema?

### DISCUSSION THREAD

**Discussion by alshdavid** [apache/arrow-rs#5633] [Q&A]
Hi, sorry for the super basic question but I am struggling getting off the ground despite the examples and documentation 🙏

I am implementing a simple IPC layer between a Rust host process and a Node.js child process and am trying to get up and running with a ping/pong request/response between the processes.

To get started I tried out Arrow using the JavaScript `avsc` npm package, which works
```javascript
const type = avsc.Type.forSchema('string')

const bytes = new Uint8Array(type.toBuffer("o w"))
console.log(bytes) // [ 6, 111, 32, 119 ]
console.log(new TextDecoder().decode(bytes)) // "o w"

const val = type.fromBuffer(Buffer.from([ 6, 111, 32, 119 ])) 
console.log(val) // "o w"
```

I'm a bit confused trying to replicate this in Rust
```rust
let mut s = StringArray::from_iter_values(&["o w"]);
println!("{:?}", s.values()); // sort of similar but not quite

let msg: Vec<u8> = s.to_bytes(); // I want to do something like this
send_to_nodejs(msg); // So I can send the data to Nodejs
```

Any help would be appreciated!

**Comment by tustvold** [comment]
I am not very familiar with the Javascript API so I am not entirely sure what you are wanting to achieve, but in arrow arrays are lists of potentially multiple elements. 

If you want to obtain a single element from a StringArray you would use `array.value(idx)` where `idx` is the index of the element you want to retrieve the value for. Alternatively you can iterate over the values in an array using `array.iter()`
